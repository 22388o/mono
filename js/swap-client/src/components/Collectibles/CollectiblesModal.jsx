import { Grid, IconButton } from '@mui/material';
import React, { useSyncExternalStore } from 'react';
import { MyModal } from '../MyModal/MyModal';
import { Close } from '@mui/icons-material';
import { NFTCard } from './NFTCard';
import { walletStore } from '../../syncstore/walletstore';

export const CollectiblesModal = ({ open, handleClose, handleItemClick=()=>{} }) => {
  const globalWallet = useSyncExternalStore(walletStore.subscribe, () => walletStore.currentState);
  const ASSET_TYPES = globalWallet.assets;
  return (
    <MyModal open={open}>
      <Grid container direction='column' spacing={1}>
        <Grid item container direction='row' width={800}>
          <Grid item xs={1}></Grid>
          <Grid item xs={10} className='flex-center flex-middle'><h3>Collectibles</h3></Grid>
          <Grid item xs={1} textAlign='right'><IconButton onClick={handleClose}><Close /></IconButton></Grid>
        </Grid>
        {/* <Grid item container direction='row'>
          <Grid item xs={3}>11 Collectibles</Grid>
          <Grid item xs={3}>All Networks</Grid>
          <Grid item xs={6} textAlign='right'>List Ordinal with PSBT</Grid>
        </Grid> */}
        <Grid item container direction='row' spacing={1.5} width={800}>
          { ASSET_TYPES.filter(asset => asset.isNFT === true && asset.balance > 0).map((card, idx) => <NFTCard handleClick={handleItemClick} key={idx} card={card} />) }
        </Grid>
      </Grid>
    </MyModal>
  );
}