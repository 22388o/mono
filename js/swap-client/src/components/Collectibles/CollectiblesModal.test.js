import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // For the "toBeInTheDocument" matcher
import { CollectiblesModal } from './CollectiblesModal';

describe('CollectiblesModal component test', () => {
  it('renders component', () => {

    const {container} = render(
      <CollectiblesModal open={true} handleClose={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders component', () => {

    const {getByText} = render(
      <CollectiblesModal open={true} handleClose={() => {}} />
    );

    const addCustomBtn = getByText('+ Add Custom Ordinal');
    fireEvent.click(addCustomBtn);

    const customHeader = getByText('Add Custom Ordinal');
    expect(customHeader).toBeInTheDocument();
  });
});
