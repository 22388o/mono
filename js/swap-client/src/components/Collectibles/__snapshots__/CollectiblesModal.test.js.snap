// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectiblesModal component test renders component 1`] = `
<div
  aria-hidden="true"
/>
`;
